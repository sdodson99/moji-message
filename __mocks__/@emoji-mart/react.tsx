type EmojiSelectEvent = {
  native: string;
};

type PickerProps = {
  onEmojiSelect: (e: EmojiSelectEvent) => void;
};

export default function Picker({ onEmojiSelect }: PickerProps) {
  const emojis = ['ðŸ’œ', 'âœ¨'];

  return (
    <>
      {emojis.map((e) => (
        <button
          key={e}
          onClick={() =>
            onEmojiSelect({
              native: e,
            })
          }
          type="button"
        >
          {e}
        </button>
      ))}
    </>
  );
}
