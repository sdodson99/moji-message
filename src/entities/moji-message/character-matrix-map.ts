const X = true;
const _ = false;

export const CharacterMatrixMap: Record<string, boolean[][] | null> = {
  A: [
    [_, X, X, X, _],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, X, X, X, X],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, _, _, _, X],
  ],
  B: [
    [X, X, X, X, _],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, X, X, X, _],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, X, X, X, _],
  ],
  C: [
    [X, X, X, X, X],
    [X, _, _, _, _],
    [X, _, _, _, _],
    [X, _, _, _, _],
    [X, _, _, _, _],
    [X, _, _, _, _],
    [X, X, X, X, X],
  ],
  D: [
    [X, X, X, X, _],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, X, X, X, _],
  ],
  E: [
    [X, X, X, X, X],
    [X, _, _, _, _],
    [X, _, _, _, _],
    [X, X, X, X, _],
    [X, _, _, _, _],
    [X, _, _, _, _],
    [X, X, X, X, X],
  ],
  F: [
    [X, X, X, X, X],
    [X, _, _, _, _],
    [X, _, _, _, _],
    [X, X, X, X, _],
    [X, _, _, _, _],
    [X, _, _, _, _],
    [X, _, _, _, _],
  ],
  G: [
    [_, X, X, X, X],
    [X, _, _, _, _],
    [X, _, _, _, _],
    [X, _, X, X, _],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [_, X, X, X, _],
  ],
  H: [
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, X, X, X, X],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, _, _, _, X],
  ],
  I: [
    [X, X, X, X, X],
    [_, _, X, _, _],
    [_, _, X, _, _],
    [_, _, X, _, _],
    [_, _, X, _, _],
    [_, _, X, _, _],
    [X, X, X, X, X],
  ],
  J: [
    [_, _, _, _, X],
    [_, _, _, _, X],
    [_, _, _, _, X],
    [_, _, _, _, X],
    [_, _, _, _, X],
    [X, _, _, _, X],
    [_, X, X, X, _],
  ],
  K: [
    [X, _, _, _, X],
    [X, _, _, X, _],
    [X, _, X, _, _],
    [X, X, _, _, _],
    [X, _, X, _, _],
    [X, _, _, X, _],
    [X, _, _, _, X],
  ],
  L: [
    [X, _, _, _, _],
    [X, _, _, _, _],
    [X, _, _, _, _],
    [X, _, _, _, _],
    [X, _, _, _, _],
    [X, _, _, _, _],
    [X, X, X, X, X],
  ],
  M: [
    [X, _, _, _, X],
    [X, X, _, X, X],
    [X, _, X, _, X],
    [X, _, X, _, X],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, _, _, _, X],
  ],
  N: [
    [X, _, _, _, X],
    [X, X, _, _, X],
    [X, _, X, _, X],
    [X, _, X, _, X],
    [X, _, X, _, X],
    [X, _, _, X, X],
    [X, _, _, _, X],
  ],
  O: [
    [_, X, X, X, _],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [_, X, X, X, _],
  ],
  P: [
    [X, X, X, X, _],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, X, X, X, _],
    [X, _, _, _, _],
    [X, _, _, _, _],
    [X, _, _, _, _],
  ],
  Q: [
    [_, X, X, X, _],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, _, _, X, X],
    [_, X, X, X, X],
  ],
  R: [
    [X, X, X, X, _],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, X, X, X, _],
    [X, _, X, _, _],
    [X, _, _, X, _],
    [X, _, _, _, X],
  ],
  S: [
    [X, X, X, X, X],
    [X, _, _, _, _],
    [X, _, _, _, _],
    [X, X, X, X, X],
    [_, _, _, _, X],
    [_, _, _, _, X],
    [X, X, X, X, X],
  ],
  T: [
    [X, X, X, X, X],
    [_, _, X, _, _],
    [_, _, X, _, _],
    [_, _, X, _, _],
    [_, _, X, _, _],
    [_, _, X, _, _],
    [_, _, X, _, _],
  ],
  U: [
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, X, X, X, X],
  ],
  V: [
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [_, X, _, X, _],
    [_, _, X, _, _],
  ],
  W: [
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, _, X, _, X],
    [X, _, X, _, X],
    [X, X, _, X, X],
    [X, _, _, _, X],
  ],
  X: [
    [X, _, _, _, X],
    [_, X, _, X, _],
    [_, _, X, _, _],
    [_, _, X, _, _],
    [_, _, X, _, _],
    [_, X, _, X, _],
    [X, _, _, _, X],
  ],
  Y: [
    [X, _, _, _, X],
    [X, _, _, _, X],
    [_, X, _, X, _],
    [_, _, X, _, _],
    [_, _, X, _, _],
    [_, _, X, _, _],
    [_, _, X, _, _],
  ],
  Z: [
    [X, X, X, X, X],
    [_, _, _, _, X],
    [_, _, _, X, _],
    [_, _, X, _, _],
    [_, X, _, _, _],
    [X, _, _, _, _],
    [X, X, X, X, X],
  ],
  '?': [
    [_, X, X, X, _],
    [X, _, _, _, X],
    [_, _, _, X, _],
    [_, _, X, _, _],
    [_, _, X, _, _],
    [_, _, _, _, _],
    [_, _, X, _, _],
  ],
  '!': [
    [_, _, X, _, _],
    [_, _, X, _, _],
    [_, _, X, _, _],
    [_, _, X, _, _],
    [_, _, X, _, _],
    [_, _, _, _, _],
    [_, _, X, _, _],
  ],
  '0': [
    [_, X, X, X, _],
    [X, _, _, _, X],
    [X, _, _, X, X],
    [X, _, X, _, X],
    [X, X, _, _, X],
    [X, _, _, _, X],
    [_, X, X, X, _],
  ],
  '1': [
    [_, _, X, _, _],
    [_, X, X, _, _],
    [X, _, X, _, _],
    [_, _, X, _, _],
    [_, _, X, _, _],
    [_, _, X, _, _],
    [X, X, X, X, X],
  ],
  '2': [
    [_, X, X, X, _],
    [X, _, _, _, X],
    [_, _, _, _, X],
    [_, _, _, X, _],
    [_, _, X, _, _],
    [_, X, _, _, _],
    [X, X, X, X, X],
  ],
  '3': [
    [_, X, X, X, _],
    [X, _, _, _, X],
    [_, _, _, _, X],
    [_, _, X, X, _],
    [_, _, _, _, X],
    [X, _, _, _, X],
    [_, X, X, X, _],
  ],
  '4': [
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [X, X, X, X, X],
    [_, _, _, _, X],
    [_, _, _, _, X],
    [_, _, _, _, X],
  ],
  '5': [
    [X, X, X, X, X],
    [X, _, _, _, _],
    [X, _, _, _, _],
    [X, X, X, X, _],
    [_, _, _, _, X],
    [X, _, _, _, X],
    [_, X, X, X, _],
  ],
  '6': [
    [_, _, _, X, _],
    [_, _, X, _, _],
    [_, X, _, _, _],
    [X, X, X, X, _],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [_, X, X, X, _],
  ],
  '7': [
    [X, X, X, X, X],
    [_, _, _, _, X],
    [_, _, _, _, X],
    [_, _, _, X, _],
    [_, _, X, _, _],
    [_, X, _, _, _],
    [X, _, _, _, _],
  ],
  '8': [
    [_, X, X, X, _],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [_, X, X, X, _],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [_, X, X, X, _],
  ],
  '9': [
    [_, X, X, X, _],
    [X, _, _, _, X],
    [X, _, _, _, X],
    [_, X, X, X, X],
    [_, _, _, X, _],
    [_, _, X, _, _],
    [_, X, _, _, _],
  ],
  ' ': [
    [_, _, _, _, _],
    [_, _, _, _, _],
    [_, _, _, _, _],
  ],
};
