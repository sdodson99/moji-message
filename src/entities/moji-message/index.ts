export { createMojiMessage } from './create-moji-message';
export type { CreateMojiMessageRequest } from './create-moji-message';
