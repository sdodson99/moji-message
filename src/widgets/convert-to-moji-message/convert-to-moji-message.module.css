.mojiMessageOutputContainer {
  min-height: 200px;

  @apply rounded bg-gray-50 border;
}
