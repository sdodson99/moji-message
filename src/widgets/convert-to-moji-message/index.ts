export { ConvertToMojiMessage } from "./convert-to-moji-message";
