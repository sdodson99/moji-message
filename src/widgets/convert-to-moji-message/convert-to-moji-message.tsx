import { useState } from 'react';
import styles from './convert-to-moji-message.module.css';
import { CreateMojiMessageRequest } from '@/entities/moji-message';
import { CopyMojiMessageButton } from '@/features/copy-moji-message';
import { CreateMojiMessageForm } from '@/features/create-moji-message';

export function ConvertToMojiMessage() {
  const [mojiMessage, setMojiMessage] = useState<string>();
  const [lastCreateMojiMessageRequest, setLastCreateMojiMessageRequest] =
    useState<CreateMojiMessageRequest>();

  const onMojiMessageCreate = (
    mojiMessage: string,
    createMojiMessageRequest: CreateMojiMessageRequest
  ) => {
    setMojiMessage(mojiMessage);
    setLastCreateMojiMessageRequest(createMojiMessageRequest);
  };

  return (
    <>
      <section>
        <CreateMojiMessageForm onMojiMessageCreate={onMojiMessageCreate} />
      </section>
      {mojiMessage ? (
        <section className="relative mt-12">
          <div className="absolute top-5 right-5">
            <CopyMojiMessageButton
              mojiMessage={mojiMessage}
              lastCreateMojiMessageRequest={lastCreateMojiMessageRequest}
            />
          </div>
          <div className={styles.mojiMessageOutputContainer}>
            <div data-testid="mojiMessageOutput" className="whitespace-pre p-5">
              {mojiMessage}
            </div>
          </div>
        </section>
      ) : null}
    </>
  );
}
