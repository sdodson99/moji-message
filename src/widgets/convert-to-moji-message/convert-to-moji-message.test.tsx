import { CreateMojiMessageRequest } from '@/entities/moji-message';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import { ConvertToMojiMessage } from './convert-to-moji-message';
import copy from 'copy-to-clipboard';
import { Mock } from 'vitest';

vi.mock('@emoji-mart/react');

vi.mock('copy-to-clipboard');
const mockCopy = copy as Mock;

async function submitCreateMojiMessageForm(request: CreateMojiMessageRequest) {
  const messageInput = screen.getByLabelText('Enter a Message');
  await userEvent.type(messageInput, request.message);

  const openMessageEmojiPickerButton = screen.getByLabelText('Choose an Emoji');
  await userEvent.click(openMessageEmojiPickerButton);

  const messageEmoji = screen.getByText(request.messageEmoji);
  await userEvent.click(messageEmoji);

  const openBackgroundEmojiPickeButton = screen.getByLabelText(
    'Choose a Background Emoji'
  );
  await userEvent.click(openBackgroundEmojiPickeButton);

  const backgroundEmoji = screen.getByText(request.backgroundEmoji);
  await userEvent.click(backgroundEmoji);

  const submitButton = screen.getByText('Convert Message');
  await userEvent.click(submitButton);
}

describe('<ConvertToMojiMessage />', () => {
  it('converts message and emojis to moji message', async () => {
    render(<ConvertToMojiMessage />);

    await submitCreateMojiMessageForm({
      message: 'hi',
      messageEmoji: 'ðŸ’œ',
      backgroundEmoji: 'âœ¨',
    });

    const mojiMessage = screen.getByTestId('mojiMessageOutput');
    expect(mojiMessage.innerHTML).toMatchSnapshot();
  });

  it('copies converted moji message w/ marketing text to clipboard when user clicks copy', async () => {
    render(<ConvertToMojiMessage />);

    await submitCreateMojiMessageForm({
      message: 'hi',
      messageEmoji: 'ðŸ’œ',
      backgroundEmoji: 'âœ¨',
    });

    const copyButton = screen.getByText('Copy');
    await userEvent.click(copyButton);

    expect(mockCopy.mock.calls[0][0]).toMatchSnapshot();
  });
});
