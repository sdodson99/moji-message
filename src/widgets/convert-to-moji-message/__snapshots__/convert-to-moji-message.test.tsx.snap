// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ConvertToMojiMessage /> > converts message and emojis to moji message 1`] = `
"âœ¨âœ¨âœ¨âœ¨âœ¨âœ¨âœ¨
âœ¨ğŸ’œâœ¨âœ¨âœ¨ğŸ’œâœ¨
âœ¨ğŸ’œâœ¨âœ¨âœ¨ğŸ’œâœ¨
âœ¨ğŸ’œâœ¨âœ¨âœ¨ğŸ’œâœ¨
âœ¨ğŸ’œğŸ’œğŸ’œğŸ’œğŸ’œâœ¨
âœ¨ğŸ’œâœ¨âœ¨âœ¨ğŸ’œâœ¨
âœ¨ğŸ’œâœ¨âœ¨âœ¨ğŸ’œâœ¨
âœ¨ğŸ’œâœ¨âœ¨âœ¨ğŸ’œâœ¨
âœ¨âœ¨âœ¨âœ¨âœ¨âœ¨âœ¨
âœ¨âœ¨âœ¨âœ¨âœ¨âœ¨âœ¨
âœ¨ğŸ’œğŸ’œğŸ’œğŸ’œğŸ’œâœ¨
âœ¨âœ¨âœ¨ğŸ’œâœ¨âœ¨âœ¨
âœ¨âœ¨âœ¨ğŸ’œâœ¨âœ¨âœ¨
âœ¨âœ¨âœ¨ğŸ’œâœ¨âœ¨âœ¨
âœ¨âœ¨âœ¨ğŸ’œâœ¨âœ¨âœ¨
âœ¨âœ¨âœ¨ğŸ’œâœ¨âœ¨âœ¨
âœ¨ğŸ’œğŸ’œğŸ’œğŸ’œğŸ’œâœ¨
âœ¨âœ¨âœ¨âœ¨âœ¨âœ¨âœ¨"
`;

exports[`<ConvertToMojiMessage /> > copies converted moji message w/ marketing text to clipboard when user clicks copy 1`] = `
"âœ¨âœ¨âœ¨âœ¨âœ¨âœ¨âœ¨
âœ¨ğŸ’œâœ¨âœ¨âœ¨ğŸ’œâœ¨
âœ¨ğŸ’œâœ¨âœ¨âœ¨ğŸ’œâœ¨
âœ¨ğŸ’œâœ¨âœ¨âœ¨ğŸ’œâœ¨
âœ¨ğŸ’œğŸ’œğŸ’œğŸ’œğŸ’œâœ¨
âœ¨ğŸ’œâœ¨âœ¨âœ¨ğŸ’œâœ¨
âœ¨ğŸ’œâœ¨âœ¨âœ¨ğŸ’œâœ¨
âœ¨ğŸ’œâœ¨âœ¨âœ¨ğŸ’œâœ¨
âœ¨âœ¨âœ¨âœ¨âœ¨âœ¨âœ¨
âœ¨âœ¨âœ¨âœ¨âœ¨âœ¨âœ¨
âœ¨ğŸ’œğŸ’œğŸ’œğŸ’œğŸ’œâœ¨
âœ¨âœ¨âœ¨ğŸ’œâœ¨âœ¨âœ¨
âœ¨âœ¨âœ¨ğŸ’œâœ¨âœ¨âœ¨
âœ¨âœ¨âœ¨ğŸ’œâœ¨âœ¨âœ¨
âœ¨âœ¨âœ¨ğŸ’œâœ¨âœ¨âœ¨
âœ¨âœ¨âœ¨ğŸ’œâœ¨âœ¨âœ¨
âœ¨ğŸ’œğŸ’œğŸ’œğŸ’œğŸ’œâœ¨
âœ¨âœ¨âœ¨âœ¨âœ¨âœ¨âœ¨

ğŸš€ Powered by Moji Message (https://emoji.seandodson.com)"
`;
